### ISSUE TITLE
Loss didn't go down when using zero optimizer with BERT pretrain

### ISSUE THREAD

**Issue by LiweiPeng** [microsoft/DeepSpeed#140]
I used zero optimizer with BERT pretrain, the loss didn't go down. However, turning off zero and using deepspeed's fp16 only, the loss went down properly. The following was what I did:

I followed the https://github.com/microsoft/DeepSpeedExamples, and added deepspeed to my BERT pretrain code. When I used similar configuration like https://github.com/microsoft/DeepSpeedExamples/blob/master/bing_bert/deepspeed_bsz16K_lamb_config.json (optimizer  is changed to adam), the pretrain looks OK. I can see the loss going down from 10.x to 5.x in less than 5000 steps. 

However, when I turned on zero optimizer in the deepspeed config file, the loss stayed at around 7.6 to 8.0, even after 20000 steps. The only difference between this one and the above working one is zero optimizer or fp16-optimizer.

I tried to use different options for adam like bias_correction true/false, eps=1e-6, 1e-8. I also tuned different learning rate. No difference was found. 

The following is my ds config file
```
{
  "zero_optimization": true,
  "train_batch_size": null,
  "train_micro_batch_size_per_gpu": 64,
  "gradient_accumulation_steps": 1,
  "gradient_clipping": 1.0,
  "steps_per_print": 100,
  "optimizer": {
    "type": "adam",
    "legacy_fusion" : false,
    "params": {
      "lr": 3e-4,
      "weight_decay": 0.01,
      "eps": 1e-6,
      "bias_correction": false
    }
  },
  "fp16": {
    "enabled": true
  },
  "wall_clock_breakdown": false
}
```

Can you help on what could be wrong?

**Comment by jeffra** [comment]
Glad it’s working now! :) Any context on the coding bug? We’re always curious about feedback to make deepspeed easier to use and help prevent bugs.

**Comment by LiweiPeng** [comment]
My coding error is at the calling of function DeepSpeedLight::backward() in deepspeed_light.py. I thought I passed True to allreduce_gradients but it turned out I passed False to it. Eventually I simplified my code and just use backward(loss), which uses default allreduce_gradients=True.
```
  deepspeed_light.py:
    def backward(self, loss, allreduce_gradients=True):
```
